// @aws-sdk/client-timestream-query/dist-es/commands/DeleteScheduledQueryCommand.js
import { getEndpointDiscoveryPlugin } from "@aws-sdk/middleware-endpoint-discovery";
import { getEndpointPlugin } from "@smithy/middleware-endpoint";
import { getSerdePlugin } from "@smithy/middleware-serde";
import { Command as $Command } from "@smithy/smithy-client";
import { SMITHY_CONTEXT_KEY, } from "@smithy/types";
import { de_DeleteScheduledQueryCommand, se_DeleteScheduledQueryCommand } from "../protocols/Aws_json1_0";
export { $Command };
export class DeleteScheduledQueryCommand extends $Command {
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, DeleteScheduledQueryCommand.getEndpointParameterInstructions()));
        this.middlewareStack.use(getEndpointDiscoveryPlugin(configuration, { clientStack, options, isDiscoveredEndpointRequired: true }));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "TimestreamQueryClient";
        const commandName = "DeleteScheduledQueryCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: (_) => _,
            outputFilterSensitiveLog: (_) => _,
            [SMITHY_CONTEXT_KEY]: {
                service: "Timestream_20181101",
                operation: "DeleteScheduledQuery",
            },
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return se_DeleteScheduledQueryCommand(input, context);
    }
    deserialize(output, context) {
        return de_DeleteScheduledQueryCommand(output, context);
    }
}
